frontend/src/old-pages/Clusters/JobSubmitDialog.tsx

```diff
@@ -145,55 +146,137 @@ function JobCostEstimate() {
   </>
 }
 
+interface JobFieldProps {
+  header: string;
+  description: string;
+  placeholder: string;
+  property: string;
+  disabled?: boolean;
+}
+
+function JobField({
+  header,
+  description,
+  placeholder,
+  property,
+  disabled=false
+}: JobFieldProps) {
+
+  const setStateIfNotEmpty = (detail: any) => {
+    !detail || detail === '' ? clearState([...jobPath, property]) : setState([...jobPath, property], detail);
+  }
```

Comment:
issue(types): you can properly type `detail` by using the types exported by the library: `InputProps.ChangeDetail`

issue(best-practice): we should always rely on `useCallback`, when applicable, for functions that we create inside the component and then pass down as props. See [here](https://reactjs.org/docs/hooks-reference.html#usecallback)